### DISCUSSION TITLE
EACCES: permission denied, mkdir '/ home / wiki'

### DISCUSSION THREAD

**Discussion by eduardoleallima** [requarks/wiki#3282] [Error / Bug Report]
Everything we are going to do on our wiki has a permission error.

**EACCES: permission denied, mkdir '/ home / wiki'**
![wiki](https://user-images.githubusercontent.com/26935908/102268643-555f7f00-3efa-11eb-9269-0ab9bc2eac55.png)

Do you know how to solve it?

Thank you very much in advance

**Comment by ghost** [comment]
You haven't provided the required info about your host! (OS, Wiki.js version, Database engine)

**Comment by eduardoleallima** [comment]
We are using Ubuntu 18.04 LTS
We followed the Wiki.js standard installation tutorial

Docker
PostgreSQL 11 (dockerized)
Wiki.js 2.x (dockerized, accessible via port 80)
Wiki.js Update Companion (dockerized)
OpenSSH with UFW Firewall preconfigured for SSH, HTTP and HTTPS

**Comment by NGPixel** [comment]
Make sure the node process has write permission to the path you specified.

**Comment by ansonphong** [comment]
I am having a related same issue, and Node does have write permission since the write operation is occurring, though it still throws this error, which is confusing for anybody trying to edit the wiki.

`EACCES: permission denied, mkdir '/var/wiki_vs_import'`

What is the significance of the `wiki_vs_import` folder?

**Reply by donoversea** [reply]
if you are running the wiki inside a docker container then you might want to change the /var to ./var which points to the container env

**Reply by andnkey** [reply]
> 

Solved for me!

**Comment by bladetin** [comment]
Just create a docker mapping, I have solved this problem

**Reply by willdkp** [reply]
Can you provide more information about how to do this?

**Reply by bladetin** [reply]
If you use docker to install wikijs, the inside and outside of docker do not communicate with each other, so you need to map the address in the container to the outside of the container (that is, the server). If you have created a Wikijs container, you need to stop the container first, and then Modify its configuration file and increase the mapping address.

You can do a search on how to modify the mapping of a container without deleting it. Reference URL：https://www.cnblogs.com/poloyy/p/13993832.html

Reference config：

{
  "StreamConfig": {},
  "State": {
    "Running": false,
    "Paused": false,
    "Restarting": false,
    "OOMKilled": false,
    "RemovalInProgress": false,
    "Dead": false,
    "Pid": 0,
    "ExitCode": 137,
    "Error": "",
    "StartedAt": "2022-06-04T15:07:41.020258166Z",
    "FinishedAt": "2022-06-05T03:40:29.183303148Z",
    "Health": null
  },
  "ID": "ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc",
  "Created": "2022-06-04T13:16:05.589484336Z",
  "Managed": false,
  "Path": "docker-entrypoint.sh",
  "Args": ["node", "server"],
  "Config": {
    "Hostname": "wiki",
    "Domainname": "",
    "User": "node",
    "AttachStdin": false,
    "AttachStdout": true,
    "AttachStderr": true,
    "ExposedPorts": { "3000/tcp": {}, "3443/tcp": {} },
    "Tty": false,
    "OpenStdin": false,
    "StdinOnce": false,
    "Env": [
      "UPGRADE_COMPANION=1",
      "DB_TYPE=postgres",
      "DB_HOST=db",
      "DB_PORT=5432",
      "DB_PASS_FILE=/etc/wiki/.db-secret",
      "DB_USER=wiki",
      "DB_NAME=wiki",
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
      "NODE_VERSION=16.15.0",
      "YARN_VERSION=1.22.18"
    ],
    "Cmd": ["node", "server"],
    "Image": "ghcr.io/requarks/wiki:2",
    "Volumes": { "/wiki/data/content": {} },
    "WorkingDir": "/wiki",
    "Entrypoint": ["docker-entrypoint.sh"],
    "OnBuild": null,
    "Labels": { "maintainer": "requarks.io" }
  },
  "Image": "sha256:aae079a1d46b84dc2101d60ac020c3101313308c413df4eb3c7cc7a83c3f52f2",
  "NetworkSettings": {
    "Bridge": "",
    "SandboxID": "714c4b3f0ad6a1c0dfdb7ab0ab14d3430801ba2c2cc07598b5e496b66b731f5c",
    "HairpinMode": false,
    "LinkLocalIPv6Address": "",
    "LinkLocalIPv6PrefixLen": 0,
    "Networks": {
      "wikinet": {
        "IPAMConfig": null,
        "Links": null,
        "Aliases": ["ffe2d57ea2bf", "wiki"],
        "NetworkID": "ee681af0296b30a1ef07c78520c9e00cb845b295561a3c0c760a94ce29d09587",
        "EndpointID": "",
        "Gateway": "",
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "MacAddress": "",
        "DriverOpts": null,
        "IPAMOperational": false
      }
    },
    "Service": null,
    "Ports": null,
    "SandboxKey": "/var/run/docker/netns/714c4b3f0ad6",
    "SecondaryIPAddresses": null,
    "SecondaryIPv6Addresses": null,
    "IsAnonymousEndpoint": false,
    "HasSwarmEndpoint": false
  },
  "LogPath": "/var/lib/docker/containers/ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc/ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc-json.log",
  "Name": "/wiki",
  "Driver": "overlay2",
  "OS": "linux",
  "MountLabel": "",
  "ProcessLabel": "",
  "RestartCount": 0,
  "HasBeenStartedBefore": true,
  "HasBeenManuallyStopped": false,
  "MountPoints": {
    "/etc/wiki/.db-secret": {
      "Source": "/etc/wiki/.db-secret",
      "Destination": "/etc/wiki/.db-secret",
      "RW": false,
      "Name": "",
      "Driver": "",
      "Type": "bind",
      "Relabel": "ro",
      "Propagation": "rprivate",
      "Spec": {
        "Type": "bind",
        "Source": "/etc/wiki/.db-secret",
        "Target": "/etc/wiki/.db-secret",
        "ReadOnly": true
      },
      "SkipMountpointCreation": false
    },
    "/wiki/data/content": {
      "Source": "",
      "Destination": "/wiki/data/content",
      "RW": true,
      "Name": "d4f79e5494af3748d8cc31b6414a87803ae139cdc8918d460eafa45a1e6305bd",
      "Driver": "local",
      "Type": "volume",
      "Spec": {},
      "SkipMountpointCreation": false
    },
    **"/home/wiki/backup": {
        "Source": "/home/wiki/backup",
        "Destination": "/home/wiki/backup",
        "RW": true,
        "Name": "",
        "Driver": "",
        "Type": "bind",
        "Propagation": "rprivate",
        "Spec": {
          "Type": "bind",
          "Source": "/home/wiki/backup",
          "Target": "/home/wiki/backup"
        },
        "SkipMountpointCreation": false
    }**
  },
  "SecretReferences": null,
  "ConfigReferences": null,
  "AppArmorProfile": "docker-default",
  "HostnamePath": "/var/lib/docker/containers/ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc/hostname",
  "HostsPath": "/var/lib/docker/containers/ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc/hosts",
  "ShmPath": "",
  "ResolvConfPath": "/var/lib/docker/containers/ffe2d57ea2bf654274d549a678e2fcabfe337a451e64c840e03c168fc74ca1fc/resolv.conf",
  "SeccompProfile": "",
  "NoNewPrivileges": false,
  "LocalLogCacheMeta": { "HaveNotifyEnabled": false }
}

**Reply by bladetin** [reply]
Note the configuration in bold black.

**Comment by Qiqiqiqiqishuo** [comment]
@eduardoleallima As the image you post above,  your "Local File System" is active in the "Storage" page. It should NOT be active, **please keep it INACTIVE to solve this problem.** 

I did my experiment in 2022/09/01(UTC+8) (follow the [Install on Ubuntu 18.04 / 20.04 / 22.04 LTS](https://docs.requarks.io/install/ubuntu) , should be the same as what you did(except the port setting), Ubuntu 18.04 LTS, PostgreSQL 11 (dockerized), 
Wiki.js 2.x (dockerized, accessible via port 3000), Wiki.js Update Companion (dockerized), OpenSSH with UFW Firewall preconfigured for SSH, HTTP and HTTPS). After install, the "Local File System" is INACTIVE in my wiki.js. And my wiki.js works fine.

As I GUESS, you activated the "Local File System" and set the Path like "/home/wiki/xxx/xxx/xxx". So ANY CHANGE would call the backup operation, which would MAKE THE PATH as you set to dump the backup-content to your disk, and the USER in the CONTAINER "wiki" has no permission to exec "mkdir /home/wiki".
So I did my TEST: Activate the "Local File System" and set the Path as "/home/wiki/backup", then I get the warning: "EACCES: permission denied, mkdir '/home/wiki'" whenever I modify my pages. Luckily, all modifications are saved correctly(though something weird happened). I enter the container wiki by "docker exec -it wiki /bin/bash" then try "mkdir /home/wiki" ---> "no permission". exec "ps -elf" ==> shows the user node run the wiki, exec "whoami" ---> "node" ==> obviously no permission to exec "mkdir /home/wiki". And node is not a sudoer?
After I DEACTIVE the "Local File System", everything works fine.

I would be very happy if my reply could help you😄. By the way, I not a native speaker, my reply may have many errors, please help me correct some of them if you like. Thank you very much in advance.

**Reply by AmIBeingObtuse** [reply]
Thank you, this worked for me.

**Reply by T342guy** [reply]
that fixed it, big thanks!
